lib/view_model/buy/buy_sell_view_model.dart

```diff
@@ -0,0 +1,357 @@
+import 'package:cake_wallet/generated/i18n.dart';
+import 'package:cake_wallet/store/app_store.dart';
+import 'package:cake_wallet/store/settings_store.dart';
+import 'package:cake_wallet/themes/theme_base.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:intl/intl.dart';
+import 'package:mobx/mobx.dart';
+
+part 'buy_sell_view_model.g.dart';
+
+class BuySellViewModel = BuySellViewModelBase with _$BuySellViewModel;
+
+abstract class BuySellViewModelBase extends WalletChangeListenerViewModel with Store {
+  BuySellViewModelBase(
+    AppStore appStore,
+  )   : _cryptoNumberFormat = NumberFormat(),
+        cryptoAmount = '',
+        fiatAmount = '',
+        cryptoCurrencyAddress = '',
+        cryptoCurrencies = <CryptoCurrency>[],
+        fiatCurrencies = <FiatCurrency>[],
+        paymentMethodState = InitialPaymentMethod(),
+        buySellQuotState = InitialBuySellQuotState(),
+        cryptoCurrency = appStore.wallet!.currency,
+        fiatCurrency = appStore.settingsStore.fiatCurrency,
+        providerList = [],
+        sortedRecommendedQuotes = ObservableList<Quote>(),
+        sortedQuotes = ObservableList<Quote>(),
+        paymentMethods = ObservableList<PaymentMethod>(),
+        settingsStore = appStore.settingsStore,
+        super(appStore: appStore) {
+    const excludeFiatCurrencies = [];
+    const excludeCryptoCurrencies = [];
+
+    fiatCurrencies =
+        FiatCurrency.all.where((currency) => !excludeFiatCurrencies.contains(currency)).toList();
+    cryptoCurrencies = CryptoCurrency.all
+        .where((currency) => !excludeCryptoCurrencies.contains(currency))
+        .toList();
+    _initialize();
+  }
+
+  @observable
+  List<CryptoCurrency> cryptoCurrencies;
+
+  @observable
+  List<FiatCurrency> fiatCurrencies;
+
+  final NumberFormat _cryptoNumberFormat;
+  late Timer bestRateSync;
+
+  List<BuyProvider> get availableBuyProviders {
+    final providerTypes = ProvidersHelper.getAvailableBuyProviderTypes(wallet.type);
+    return providerTypes
+        .map((type) => ProvidersHelper.getProviderByType(type))
+        .where((provider) => provider != null)
+        .cast<BuyProvider>()
+        .toList();
+  }
+
+  List<BuyProvider> get availableSellProviders {
+    final providerTypes = ProvidersHelper.getAvailableSellProviderTypes(wallet.type);
+    return providerTypes
+        .map((type) => ProvidersHelper.getProviderByType(type))
+        .where((provider) => provider != null)
+        .cast<BuyProvider>()
+        .toList();
+  }
+
+  @override
+  void onWalletChange(wallet) {
+    cryptoCurrency = wallet.currency;
+  }
+
+  bool get isDarkTheme => settingsStore.currentTheme.type == ThemeType.dark;
+
+  double get amount {
+    final formattedFiatAmount = double.tryParse(fiatAmount) ?? 200.0;
+    final formattedCryptoAmount =
+        double.tryParse(cryptoAmount) ?? (cryptoCurrency == CryptoCurrency.btc ? 0.001 : 1);
```

Comment:
We discussed this. Most providers do not provide a quote if the fiat amount is less than 100. For crypto, I set it to 0.001 since some providers can’t afford to trade with 1 BTC. Other altcoins can be 1, I guess. In any case, we need to provide an amount to get a valid quote.